pages/api/augment-request.ts

```diff
@@ -38,148 +77,156 @@ function castOracleNameForOOUi(oracleType: string): string {
     case "SkinnyOptimisticOracle":
       return "Skinny";
     default:
-      return "";
+      throw new Error("Unable to cast oracle name for OO UI: " + oracleType);
   }
 }
 
-async function constructOptimisticOraclesByChain(chainId: SupportedChainIds) {
-  const optimisticOracle = (await constructContract(
-    chainId,
-    "OptimisticOracle"
-  )) as OptimisticOracleEthers;
-  const optimisticOracleV2 = (await constructContract(
-    chainId,
-    "OptimisticOracleV2"
-  )) as OptimisticOracleV2Ethers;
-
-  // Only mainnet has a SkinnyOptimisticOracle so only construct it here. All other chains of interest have OOv1 and V2.
-  const skinnyOptimisticOracle =
-    chainId == 1
-      ? ((await constructContract(
-          chainId,
-          "SkinnyOptimisticOracle"
-        )) as SkinnyOptimisticOracleEthers)
-      : null;
-  return [optimisticOracle, optimisticOracleV2, skinnyOptimisticOracle];
+type LookupTable<E> = Record<string, Record<number, E>>;
+function createLookupTable<Event extends { time: number; identifier: string }>(
+  events: Event[],
+  table: LookupTable<Event> = {}
+): LookupTable<Event> {
+  return events.reduce((table, event) => {
+    const identifier = event.identifier.toLowerCase();
+    const time = event.time;
+    if (identifier && table[identifier] === undefined) table[identifier] = {};
+    if (time && table[identifier][time] === undefined)
+      table[identifier][time] = event;
+    return table;
+  }, table);
 }
 
-async function getOptimisticOraclePriceRequestEventsByChainId(
+async function getVotingPriceRequestAdded(
+  contractType: VotingType,
   chainId: SupportedChainIds
-) {
-  const chainOptimisticOracles = await constructOptimisticOraclesByChain(
-    chainId
+): Promise<CommonEventData[]> {
+  const contract = await constructContract(chainId, contractType);
+  const events = await contract.queryFilter(
+    contract.filters.PriceRequestAdded(),
+    getFromBlock(contractType as string, chainId as number)
   );
-
-  const requests = await Promise.all(
-    chainOptimisticOracles.map((optimisticOracle) =>
-      optimisticOracle
-        ? optimisticOracle.queryFilter(optimisticOracle.filters.RequestPrice())
-        : null
+  return events.map((event) =>
+    ss.create(
+      {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+        transactionHash: event?.transactionHash || "rolled",
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+        identifier: event?.args?.identifier,
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
```

Comment:
the linter really hated this, even though its pretty safe. we just need to drill into an unknown object, and then use superstruct to validate the end result, this means we dont need any explicit casting, so i disabled all the linter errors

Reply:
its a little more specific this way, rather than turning it off for the entire file, its only for these lines, but good to know that is an option 😄